test/e2e_env/kubernetes/k8s_api_bypass/k8s_api_bypass.go

```diff
@@ -72,9 +83,14 @@ spec:
 
 		// then communication with API Server still works
 		Eventually(func(g Gomega) {
-			stdout, _, err := kubernetes.Cluster.Exec(namespace, clientPodName, "demo-client", "bash", "-c", cmd)
+			stdout, _, err := client.CollectRawResponse(
+				kubernetes.Cluster, "demo-client", "https://kubernetes.default.svc/api",
+				client.FromKubernetesPod(namespace, "demo-client"),
+				client.WithCACert(caCert),
+				client.WithHeader("Authorization", fmt.Sprintf("Bearer %s", token)),
+			)
 			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(stdout).To(Equal("200"))
+			g.Expect(stdout).To(ContainSubstring(`"kind": "APIVersions"`))
```

Comment:
Maybe this deserves a comment? It's not clear to me why this means the test was successful